subsys/power/device_pm.c

```diff
@@ -112,6 +112,10 @@ static int device_pm_request(const struct device *dev,
 		}
 	}
 
+	if (k_is_pre_kernel()) {
```

Comment:
I assume these changes imply that `pm_work_handler` will be never run in pre_kernel stage. Let's discuss that later more in detail.

Reply:
yep, I tested with the blinky sample (last commit) and it seems to be working